PolyEngine/Core/Src/Optional.hpp

```diff
@@ -0,0 +1,188 @@
+#pragma once
+
+#include "Defines.hpp"
+
+namespace Poly {
+	template<typename V>
+	class Optional {
+	public:
+		using ValueType = V;
+	public:
+		Optional() : initialized(false), dummy{} {}
+		Optional(const V&  value) : initialized(true), value_storage(value) {}
+		Optional(      V&& value) : initialized(true), value_storage(std::move(value)) {}
```

Comment:
I find aligned lines to be of tremendous help when it comes to code readability and how easy it is to spot a mistake. I try to keep a good balance as not to format code in a weird way, but sometimes there's a border case like this. Should I change this?
P.S. find the error:
```
Optional<V&> Get(const K&  key) { return this->get(key); }
Optional<V&> Get(      K&  key) { return this->get(std::move(key)); }
```
(I just fixed it :P)